src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -18,8 +18,14 @@
 
 logger = get_logger(__name__)
 
+# This default cloud name. If you use DummyCli(random_config_dir=True), use get_cloud_config_file() instead.
 CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
```

Comment:
The original implementation is wrong. 

`get_default_cli` uses `GLOBAL_CONFIG_DIR` for `knack.cli.CLI.__init__`'s `config_dir` argument:

https://github.com/Azure/azure-cli/blob/63121edc73fa103658b3184d3d9d0328f43f330a/src/azure-cli-core/azure/cli/core/__init__.py#L902-L903

But, if `knack.cli.CLI.__init__`'s `config_dir` doesn't use `GLOBAL_CONFIG_DIR`, but another location, like what has been done in #25689:

https://github.com/Azure/azure-cli/blob/b223dd9d4379d5c633a9f4dba4941ae791531126/src/azure-cli-core/azure/cli/core/mock.py#L28-L29

Then the location of `clouds.config` will still be under `GLOBAL_CONFIG_DIR`, this defeats the purpose of overriding `config_dir`.

Reply:
`profile._storage` is an `azure.cli.core._session.ACCOUNT` which is a persisted JSON. It can easily be replaced by an in-memory `dict` during tests, but `cloud.config` is an INI file, which introduces inconsistency. See https://github.com/Azure/azure-cli/pull/14436#issuecomment-1514207629